sycl/source/detail/platform_impl.cpp

```diff
@@ -26,6 +27,50 @@ namespace detail {
 
 using PlatformImplPtr = std::shared_ptr<platform_impl>;
 
+// Return true if the given device 'Dev' matches with any filter
+bool isForcedDevice(detail::device_filter_list *FilterList,
+                    const plugin &Plugin, const RT::PiDevice &Dev,
+                    unsigned int Index) {
+  backend Backend = Plugin.getBackend();
+  info::device_type Type;
+  Plugin.call<PiApiKind::piDeviceGetInfo>(
+      Dev, pi::cast<RT::PiDeviceInfo>(info::device::device_type), sizeof(Type),
+      &Type, nullptr);
```

Comment:
Any reason why you get the device type by calling a plugin API piDeviceGetInfo instead of  ?
`info::device_type` Type = Dev.get_info<info::device::device_type>();

Reply:
done